### PR TITLE
typing for rest.py

### PR DESCRIPTION
**Pull request by hackaugusto**
I added the type for the RaidenService so that jedi can jump around the definitions. That caught two bugs in the connection manager endpoint.

### CODE REVIEW

File: raiden/api/rest.py [hunk 1]
```diff
@@ -506,9 +527,7 @@ def get_our_address(self):
         return api_response(result=dict(our_address=to_checksum_address(self.raiden_api.address)))
 
     def register_token(
-        self,
-        registry_address: typing.TokenNetworkRegistryAddress,
-        token_address: typing.TokenAddress,
+        self, registry_address: TokenNetworkRegistryAddress, token_address: TokenAddress
     ):
```
Comment (palango) [inline raiden/api/rest.py]: missing return type

File: raiden/api/rest.py [hunk 2]
```diff
@@ -303,8 +319,13 @@ class APIServer(Runnable):  # pragma: no unittest
     _api_prefix = "/api/1"
 
     def __init__(
-        self, rest_api, config, cors_domain_list=None, web_ui=False, eth_rpc_endpoint=None
-    ):
+        self,
+        rest_api: "RestAPI",
+        config: Dict[str, Any],
+        cors_domain_list=None,
```
Comment (palango) [inline raiden/api/rest.py]: types missing here

File: raiden/api/rest.py [hunk 3]
```diff
@@ -548,10 +567,10 @@ def register_token(
...
         self,
-        token_address: typing.TokenAddress,
-        to: typing.Address,
-        value: typing.TokenAmount,
-        contract_method: str,
+        token_address: TokenAddress,
+        to: Address,
+        value: TokenAmount,
+        contract_method: MintingMethod,
     ):
```
Comment (palango) [inline raiden/api/rest.py]: missing return type

File: raiden/api/rest.py [hunk 4]
```diff
@@ -662,11 +681,11 @@ def open(
...
-        registry_address: typing.TokenNetworkRegistryAddress,
-        token_address: typing.TokenAddress,
-        funds: typing.TokenAmount,
-        initial_channel_target: int = None,
-        joinable_funds_target: float = None,
+        registry_address: TokenNetworkRegistryAddress,
+        token_address: TokenAddress,
+        funds: TokenAmount,
+        initial_channel_target: int = 3,
+        joinable_funds_target: float = 0.4,
```
Comment (rakanalh) [inline raiden/api/rest.py]: You can use `settings.DEFAULT_JOINABLE_FUNDS_TARGET` and `settings.DEFAULT_INITIAL_CHANNEL_TARGET` for these values

File: raiden/api/rest.py [hunk 5]
```diff
@@ -718,20 +733,23 @@ def get_connection_managers_info(self, registry_address: typing.TokenNetworkRegi
...
             )
+            connection_manager = None
 
-            try:
-                connection_manager = self.raiden_api.raiden.connection_manager_for_token_network(
-                    token_network_address
-                )
-            except InvalidBinaryAddress:
-                connection_manager = None
+            if token_network_address is not None:
```
Comment (rakanalh) [inline raiden/api/rest.py]: I am assuming that this is a solution to one of the two bugs you mentioned in the description. Can't spot the other one though.

Comment (hackaugusto) [reply raiden/api/rest.py]: You did, it is in your previous comment: https://github.com/raiden-network/raiden/pull/4724#discussion_r320187088 , the value `None` is invalid.

File: raiden/api/rest.py (hunk 4 above)
Comment (hackaugusto) [reply raiden/api/rest.py]: Yes, this is unrelated to this PR. Can it be done in a follow up?

File: raiden/api/rest.py (hunk 3 above)
Comment (hackaugusto) [reply raiden/api/rest.py]: Yes, I don't know the return type of the `rest` API and preferred to leave it without an annotation. I just wanted to add the `RaidenService` with this PR.

File: raiden/api/rest.py (hunk 4 above)
Comment (rakanalh) [reply raiden/api/rest.py]: Sure

**Comment by pirapira** [comment]
> Since you're fixing types, as @palango suggested the return types are missing from most of the function signatures.

@nlsdfnbch What do you suggest from the devops perspective?  I would prefer merging smaller PRs often.

File: raiden/api/rest.py (hunk 3 above)
Comment (palango) [reply raiden/api/rest.py]: Fine for me.

**Pull request merged**